src/util/async/AnyExecutionContext.hpp

```diff
@@ -43,16 +43,41 @@ class AnyExecutionContext {
     /**
      * @brief Construct a new type-erased Execution Context object
      *
+     * @note Stores the Execution Context by reference.
+     *
+     * @tparam CtxType The type of the execution context to wrap
+     * @param ctx The execution context to wrap
+     */
+    template <typename CtxType>
+    /* implicit */
+    AnyExecutionContext(CtxType&& ctx)
```

Comment:
Could you declare these constructors with `CtxType&` and `CtxType&&`? Or it will not work?

Reply:
Hm, why couldn't you forward to `Model` whatever you got here? Then it will be possible to pass `ctx` as rvalue reference or lvalue.